distributed/shuffle/shuffle.py

```diff
@@ -63,10 +63,18 @@ def rearrange_by_column_p2p(
     npartitions = npartitions or df.npartitions
     token = tokenize(df, column, npartitions)
 
+    empty = df._meta.copy()
+    for c, dt in empty.dtypes.items():
+        if dt == object:
+            empty[c] = empty[c].astype(
+                "string"
+            )  # TODO: we fail at non-string object dtypes
+    empty[column] = empty[column].astype("int64")  # TODO: this shouldn't be necesssary
```

Comment:
TODO: This was weird, but is weird separately from this PR

Reply:
I would love to see a dedicated test for this, if there isn't one already

Reply:
I actually thing we should have both. A bug report to get the issue fixed and a test case that ensures that we actually hotfixed this bug.

If I remove the int cast, `test_basic` actually fails but I am having a very hard time figuring out what is going on. If we remove the string conversion, nothing fails. How would we know when this is safe to be removed?